Format comments with different opening in different manner

topecongiro opened the pull request:
This PR closes #691.
However, TBH I am not sure whether how I deal with this issue is ideal.
Introducing `enum CommentStyle` seems overkill after looking back this PR.
I really appreciate any feedbacks.
Thank you in advance!

nrc commented on src/comment.rs:
```diff
@@ -47,39 +159,44 @@ pub fn rewrite_comment(orig: &str,
+    if rest.is_empty() {
+        Some(first_group_str)
+    } else {
+        rewrite_comment(&rest, block_style, shape, config).map(|rest_str| {
```
Rather than have `rewrite_comment` and `rewrite_comment_inner` where the latter recurses, could we have a `rewrite_comment` function which calls an `identify_comment` where `identify_comment` is recursive?

topecongiro commented:
Thank you! Rebased reflecting the feedbacks.
1. renamed `to_pattern` to `line_with_same_comment_style`.
2. created `identify_comment` and made it recursive.

topecongiro commented on src/comment.rs:
```diff
@@ -47,39 +163,52 @@ pub fn rewrite_comment(orig: &str,
+                    -> Option<String> {
+    let style = comment_style(orig, false);
+    let first_group = orig.lines()
+        .take_while(|l| style.line_with_same_comment_style(l, orig, false))
```
I was wondering if rust has a function equivalent to [`scan`](http://hackage.haskell.org/package/base-4.9.1.0/docs/Prelude.html#v:span) in haskell.

nrc replied:
I don't think there is one - there might be one in Itertools, but I prefer not to link it and use simpler functions.

nrc replied:
You might be able to use [partition](https://doc.rust-lang.org/nightly/std/iter/trait.Iterator.html#method.partition)

nrc commented:
Thanks for the changes!
